docs/user-guide/projects.md

```diff
@@ -144,6 +144,15 @@ spec:
   ...
 ```
 
+Argo CD will use the policies defined in the AppProject roles while authorizing users actions. To determine which role a given users is associated with, it will dynamically create groups based on the role name in runtime. The project definition above will generate the following Casbin RBAC rules:
+
+```
+    p, proj:sample-test-project:some-role, applications, *, *, allow
+    g, some-user, proj:sample-test-project:some-role
+```
+
+_Note 1_: It is very important that policy roles follow the pattern `proj:<project-name>:<role-name>` or they won't be effective during the Argo CD authorization process.
+_Note 2_: The example above used `applications` as the resource for the policy definition. However all resources (`applicationsets`, `repositories`, etc) can be defined as AppProject policies. See the [RBAC documentation](../operator-manual/rbac.md) for more information about the possible resources.
```

Comment:
I'd suggest saying "all **project-scoped** resources can be defined..." since stuff like accounts and projects themselves can't be managed via this RBAC (afaik).

Reply:
updated